### PR TITLE
Fix/isolated modules

### PR DESCRIPTION
**Pull request by literat**
* `export const enum` do not work well with `isolatedModules` options
* use rather object with named values
* we should do not export const enums ever in any library (for
   internal use it works well)
* @see: https://www.typescriptlang.org/tsconfig#isolatedModules
* @see: https://ncjamieson.com/dont-export-const-enums/
* @see: https://fettblog.eu/tidy-typescript-avoid-enums/

### CODE REVIEW

File: src/types/CookieConsentManager.ts [hunk 1]
```diff
@@ -1,24 +1,25 @@
-import { CookieConsentCategory } from './CookieConsentCategory';
+import { CookieConsentCategory, DisplayMode } from '../constants';
 import { VanillaCookieConsent } from './vanilla-cookieconsent';
 
+export type Values<T> = T[keyof T];
+
+export type CookieConsentCategoryValues = Values<typeof CookieConsentCategory>;
+
```
Comment (OndraM) [inline src/types/CookieConsentManager.ts:7]: The purpose of this is to workaround for not using enums, right?

So basically we will have to take care of variables which are basically "enums" in a two different ways, depending on whether the type is part of exported public API (so you must use generics like this, ie. emulate enum behavior) or whether they are used only internally (when you use "real" enums")? That is quite a confusing... 🤔

Comment (literat) [reply src/types/CookieConsentManager.ts:7]: Well, not so far confusing. This is not the first time I encounter some problem with enums. For example iterating over enum and getting values from them is a task from hell (search something like "how to use enum as index key type in typescript").

So my opinion is to ban enums for type safety reasons. Not just that we are developing a library but also for sake of our souls. It is better not to think about whether I am using enum internally or not. Just do not use them everywhere.

**Comment by literat** [comment]
It has been confirmed that this is working solution.

File: examples/webpack-with-typescript/index.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// eslint-disable-next-line import/no-unresolved
+import LmcCookieConsentManager, { DisplayMode } from '@lmc-eu/cookie-consent-manager';
+
+const ccmArgs = {
+  displayMode: DisplayMode.SOFT,
+  config: {
+      cookie_name: 'lmc_ccm_example',
+      force_consent: true,
```
Comment (OndraM) [inline examples/webpack-with-typescript/index.ts:8]: Please remove overriding force_consent from the examples... There is already the displayMode being used, so either use it as soft or force, but don't combine it with force_consent in the examples. :)

**Review by OndraM — CHANGES_REQUESTED** [review]

File: src/types/CookieConsentManager.ts [hunk 3]
```diff
@@ -1,24 +1,25 @@
-import { CookieConsentCategory } from './CookieConsentCategory';
+import { CookieConsentCategory, DisplayMode } from '../constants';
 import { VanillaCookieConsent } from './vanilla-cookieconsent';
 
+export type Values<T> = T[keyof T];
```
[the remark this answers is not in the document]
Comment (literat) [reply src/types/CookieConsentManager.ts:4]: Here it is exported for internal use only. You must look at what is exported from the package main entry point which is `src/LmcCookieConsentManager.ts`

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+name: Build
```
[the remark this answers is not in the document]
Comment (literat) [reply .github/workflows/build.yaml]: Renamed, sounds reasonable.

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+on:
+  push:
+  schedule:
+    - cron: '0 9 * * *' # every day at 9:00
+
+jobs:
+  test:
+    name: Build
```
[the remark this answers is not in the document]
Comment (literat) [reply .github/workflows/build.yaml]: Renamed.

File: examples/webpack-with-typescript/index.ts (hunk 2 above)
Comment (literat) [reply examples/webpack-with-typescript/index.ts]: Removed.

**Review by OndraM — APPROVED** [review]

**Pull request merged**